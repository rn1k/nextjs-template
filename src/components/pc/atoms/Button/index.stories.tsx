import React from 'react';

import { storiesOf } from '@storybook/react';
// import { withInfo } from '@storybook/addon-info';
// import { withKnobs, text, boolean } from '@storybook/addon-knobs';
// import { action } from '@storybook/addon-actions';

import Button from './';

storiesOf('pc/atoms/Button', module)
    .add('Example', () => <Button>sample text</Button>)
    .add('round', () => <Button round={true}>primary</Button>);
