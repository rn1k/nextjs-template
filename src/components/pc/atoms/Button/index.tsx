import React from 'react';
import styled, { css } from 'styled-components';

interface Props {
    as?: any;
    round?: boolean;
    href?: string;
    onClick?: (event: React.MouseEvent<HTMLAnchorElement>) => void;
}
const Component = styled('a')<{
    round?: boolean;
}>`
    display: inline-block;
    padding: 8px 10px;
    margin: 0 auto;
    border-radius: 4px;
    box-sizing: border-box;
    background: #fff;
    text-align: center;
    text-decoration: none;
    border: solid 1px #888;
    ${props => {
        let styles = css``;
        if (props.round) {
            styles = css`
                padding: 8px 24px;
                border-radius: 24px;
            `;
        }
        return styles;
    }}
`;

const Button: React.FC<Props> = (props): React.ReactElement => {
    return (
        <>
            <Component
                onClick={props.onClick}
                round={props.round}
                href={props.href}
                as={props.as}
            >
                {props.children}
            </Component>
        </>
    );
};

export default Button;
