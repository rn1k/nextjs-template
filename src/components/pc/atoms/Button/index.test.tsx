import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Button from './index';

describe('Button', () => {
    describe('スナップショットテスト', () => {
        describe('round', () => {
            it('true', () => {
                const tree = renderer.create(<Button round={true} />);
                expect(tree).toMatchSnapshot();
            });
        });
    });
});
