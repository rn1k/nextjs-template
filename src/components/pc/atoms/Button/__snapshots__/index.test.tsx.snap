// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button スナップショットテスト round true 1`] = `
.c0 {
  display: inline-block;
  padding: 8px 10px;
  margin: 0 auto;
  border-radius: 4px;
  box-sizing: border-box;
  background: #fff;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: solid 1px #888;
  padding: 8px 24px;
  border-radius: 24px;
}

<a
  className="c0"
/>
`;
